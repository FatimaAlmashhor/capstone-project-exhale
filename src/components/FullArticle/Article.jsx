import React from 'react';
import { useTranslation } from 'react-i18next';
import { useParams } from 'react-router-dom';
import { articles } from '../../services/fakeArticleService';

const Article = () => {
  const { t } = useTranslation();
  const { id } = useParams();

  const article = articles.find((p) => p.id === Number(id));
  let articleData;

  if (article) {
    articleData = (
      <div>
        <section className="blog ">
          <div className="container px-5 py-24 mx-auto">
            <div className="flex flex-wrap sm:-m-4 -mx-4 -mb-10 -mt-4">
              <div className="p-5  md:mb-0 mb-6 flex flex-col justify-center items-center  mx-auto">
                {article.articleImg && (
                  <div
                    className="bg-gray-300 h-56 w-full rounded-lg shadow-md bg-cover bg-center "
                    style={{ backgroundImage: `url(${article.articleImg}` }}
                  />
                )}
                <div className="  bg-white -mt-10 shadow-lg rounded-lg overflow-hidden p-5">
                  <div className="header-content inline-flex">
                    <div className="category-badge flex-1  h-4 w-4 m rounded-full m-1 bg-green-100">
                      <div className="h-2 w-2 rounded-full m-1 bg-green-500" />
                    </div>
                    <div className="category-title flex-1 text-sm text-gray-400">
                      {article.date}
                    </div>
                  </div>
                  <div className="title-post text-3xl">
                    <h2 className="text-blue-500 ">{article.title}</h2>
                  </div>
                  <div className="summary-post text-base text-justify text-1xl ">
                    <p>{article.text}</p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    );
  } else {
    articleData = <h2>{t('Sorry. Article does not exist')}</h2>;
  }

  return (
    <div>
      <div>{articleData}</div>
    </div>
  );
};

export default Article;
