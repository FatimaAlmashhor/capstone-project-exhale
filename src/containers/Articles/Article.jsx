import React from 'react';
import FullArticle from '../../components/FullArticle/Article';

const Article = (artId) => {
  return (
    <div className="bg-background-secondary">
      <FullArticle id={artId} />
    </div>
  );
};

export default Article;
