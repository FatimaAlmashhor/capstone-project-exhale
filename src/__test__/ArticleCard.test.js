import React from 'react';
import 'react-router-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter as  Router} from 'react-router-dom'
import Article  , {HorizontalArticleCard} from '../components/ArticleCard';
it('renders correctly', () => {
  const tree = renderer
    .create(<Router><Article id={ 1 } title = {' '} paragraph = {' '} /></Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
it('renders correctly', () => {
  const tree = renderer
    .create(<Router><HorizontalArticleCard index={ 1 } title = {' '} paragraph = {' '}/></Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})