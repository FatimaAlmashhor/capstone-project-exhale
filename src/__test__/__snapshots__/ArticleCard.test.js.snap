// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="blog "
>
  <div
    className="container px-5 py-12 mx-auto"
  >
    <div
      className="flex flex-wrap -mx-4 -mt-4 -mb-10 sm:-m-4"
    >
      <div
        className="flex flex-col items-center justify-center p-5 mx-auto mb-6 md:mb-0"
      >
        <div
          className="w-full h-56 bg-gray-300 bg-center bg-cover shadow-md "
          style={
            Object {
              "backgroundImage": "url( ",
            }
          }
        />
        <div
          className="p-5 -mt-10 overflow-hidden bg-background-primary rounded-lg shadow-sm "
        >
          <div
            className="inline-flex header-content"
          >
            <div
              className="flex-1 text-sm text-gray-400 category-title"
            >
              11/24/2020
            </div>
          </div>
          <div
            className="text-lg title-post"
          >
            <h2
              className="text-textColor-primary "
            >
               
            </h2>
          </div>
          <div
            className="text-textColor-secondary text-justify summary-post text-sm "
          >
            <p>
               
            </p>
            <button
              className="block p-2 px-2 mt-4 text-sm text-white bg-green-400 rounded hover:bg-green-300"
              type="button"
            >
              <li
                className="list-none"
              >
                <a
                  href="//1"
                  onClick={[Function]}
                >
                  Read More
                </a>
              </li>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders correctly 2`] = `
<section
  className="w-full bg-transparent blog"
>
  <div
    className="container w-full px-5 py-12 mx-auto"
  >
    <div
      className="flex-wrap -mx-4 -mt-4 -mb-10 sm:-m-4 hover:shadow-lg"
    >
      <div
        className="flex flex-col m-auto lg:flex-row"
      >
        <div
          className="flex content-center w-full h-56 bg-gray-300 bg-center bg-cover shadow-sm lg:w-2/3 lg:h-auto "
          style={
            Object {
              "backgroundImage": "url( ",
            }
          }
        />
        <div
          className="w-auto rounded overflow-hidden border hover:shadow-lg"
        >
          <div
            className="p-6 m-auto -mt-10 overflow-hidden bg-background-primary rounded-lg shadow-sm "
          >
            <div
              className="header-content "
            >
              <div
                className="flex-1 w-full mt-10 text-sm text-gray-400 category-title"
              >
                11/24/2020
              </div>
            </div>
            <div
              className="mt-5 text-lg title-post"
            >
              <h2
                className="text-textColor-primary "
              >
                 
              </h2>
            </div>
            <div
              className="pt-2 text-base text-justify text-gray-500 summary-post text-md "
            >
              <p>
                 
              </p>
              <div
                className="mt-2"
              >
                <a
                  className="text-sm text-blue-400 transition duration-150 ease-in-out hover:text-blue-900"
                  href="/articles/1"
                  onClick={[Function]}
                >
                  Read more
                  <svg
                    aria-hidden="true"
                    className="inline-block w-2 h-2 ml-1 stroke-current stroke-2 chev"
                    fill="none"
                    viewBox="0 0 10 10"
                  >
                    <g
                      fillRule="evenodd"
                    >
                      <path
                        d="M1 1l4 4-4 4"
                      />
                    </g>
                  </svg>
                  <svg
                    aria-hidden="true"
                    className="hidden w-2 h-2 ml-1 stroke-current stroke-2 arr hover:inline-block"
                    fill="none"
                    viewBox="0 0 10 10"
                  >
                    <g
                      fillRule="evenodd"
                    >
                      <path
                        d="M0 5h7"
                      />
                      <path
                        d="M4 1l4 4-4 4"
                      />
                    </g>
                  </svg>
                </a>
                
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
